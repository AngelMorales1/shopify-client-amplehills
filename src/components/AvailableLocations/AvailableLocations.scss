@import 'settings', 'src/styles/utils';

.AvailableLocations {
  &--yellow {
    @extend .bg-yellow;
  }

  &--light-yellow {
    @extend .bg-light-yellow;
  }

  &--mustard {
    @extend .bg-deep-yellow;
  }

  &--blue {
    @extend .bg-sky-blue;
  }

  &--light-turquoise {
    @extend .bg-light-turquoise;
  }

  &--pink {
    @extend .bg-light-pink;
  }

  &--light-pink {
    @extend .bg-baby-pink;
  }

  &--light-peach {
    @extend .bg-light-peach;
  }

  &--light-orange {
    @extend .bg-light-orange;
  }

  &--peach-pink {
    @extend .bg-peach-pink;
  }

  &--yellow-green {
    @extend .bg-yellow-green;
  }

  &--light-yellow-green {
    @extend .bg-light-yellow-green;
  }

  padding: spacing(16) spacing(12);

  @media ($medium-up) {
    padding: spacing(16) spacing(16);
  }

  &__button-container {
    flex-direction: column;

    @media ($medium-up) {
      flex-direction: row;
    }
  }
}
