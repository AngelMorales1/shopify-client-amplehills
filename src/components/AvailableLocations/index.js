import React from 'react';
import PropTypes from 'prop-types';
import get from 'utils/get';
import cx from 'classnames';
import { Button } from 'components/base';

import styles from './AvailableLocations.scss';

const AvailableLocations = ({ flavor, block, z, setRef, drip, upperDrip }) => {
  const availableLocations = get(flavor, 'availableLocations', []);
  const fields = get(block, 'fields', {});
  const blockTitle = get(fields, 'title', '');
  const backgroundColor = get(fields, 'backgroundColor', 'white');

  return (
    <div
      ref={refBlock => setRef(refBlock)}
      style={{ zIndex: z }}
      className={cx(
        styles['AvailableLocations'],
        styles[`AvailableLocations--${backgroundColor}`],
        {
          drip: drip,
          'upper-drip': upperDrip
        }
      )}
    >
      <h2 className="block-headline center mb3">{blockTitle}</h2>
      <div className="w100 flex justify-center items-center">
        <div
          className={cx(
            styles['AvailableLocations__button-container'],
            'flex flex-wrap justify-center items-center'
          )}
        >
          {availableLocations.map(location => {
            const fields = get(location, 'fields', {});
            const locationTitle = get(fields, 'title', '');
            const slug = get(fields, 'slug', '');
            const id = get(location, 'sys.id', '');

            return (
              <Button
                key={id}
                className={cx(
                  styles['AvailableLocations__button'],
                  'uppercase inline-flex m1'
                )}
                variant="primary-small"
                color="madison-blue"
                label={locationTitle}
                to={`/location/${slug}`}
              />
            );
          })}
        </div>
      </div>
    </div>
  );
};

AvailableLocations.propTypes = {
  flavor: PropTypes.shape({
    availableLocations: PropTypes.array,
    contentBlocks: PropTypes.array,
    id: PropTypes.string,
    image: PropTypes.string,
    label: PropTypes.string,
    slug: PropTypes.string,
    title: PropTypes.string
  }),
  z: PropTypes.number,
  setRef: PropTypes.func,
  block: PropTypes.shape({
    fields: PropTypes.shape({
      contentType: PropTypes.string,
      drip: PropTypes.bool,
      title: PropTypes.string
    }),
    sys: PropTypes.shape({
      id: PropTypes.string
    })
  })
};

AvailableLocations.defaultProps = {
  flavor: {
    availableLocations: [],
    contentBlocks: [],
    id: '',
    image: '',
    label: '',
    slug: '',
    title: ''
  },
  z: 1,
  setRef: () => {},
  block: {
    fields: {
      contentType: '',
      drip: false,
      title: ''
    },
    sys: {
      id: ''
    }
  }
};

export default AvailableLocations;
