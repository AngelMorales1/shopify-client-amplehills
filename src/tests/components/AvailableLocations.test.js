import React from 'react';
import { shallow } from 'enzyme';

import AvailableLocations from 'components/AvailableLocations';

it('renders without data', () => {
  const component = shallow(<AvailableLocations />);

  expect(component).toMatchSnapshot();
});

it('renders with data', () => {
  const component = shallow(
    <AvailableLocations
      flavor={{
        availableLocations: [],
        contentBlocks: [],
        id: 'foo',
        image: 'foo',
        label: 'foo',
        slug: 'foo',
        title: 'foo'
      }}
      block={{
        fields: {
          contentType: 'foo',
          drip: false,
          title: 'foo'
        },
        sys: {
          id: 'foo'
        }
      }}
    />
  );

  expect(component).toMatchSnapshot();
});
