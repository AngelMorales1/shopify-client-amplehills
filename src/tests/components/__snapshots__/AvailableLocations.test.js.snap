// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AvailableLocations
    block={
        Object {
            "fields": Object {
              "contentType": "foo",
              "drip": false,
              "title": "foo",
            },
            "sys": Object {
              "id": "foo",
            },
          }
    }
    flavor={
        Object {
            "availableLocations": Array [],
            "contentBlocks": Array [],
            "id": "foo",
            "image": "foo",
            "label": "foo",
            "slug": "foo",
            "title": "foo",
          }
    }
    setRef={[Function]}
    z={1}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          className="block-headline center mb3"
>
          foo
</h2>,
        <div
          className="flex w100 items-center justify-center"
>
          
</div>,
      ],
      "className": "bg-light-yellow",
      "style": Object {
        "zIndex": 1,
      },
    },
    "ref": [Function],
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "foo",
          "className": "block-headline center mb3",
        },
        "ref": null,
        "rendered": "foo",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [],
          "className": "flex w100 items-center justify-center",
        },
        "ref": null,
        "rendered": Array [],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="block-headline center mb3"
>
            foo
</h2>,
          <div
            className="flex w100 items-center justify-center"
>
            
</div>,
        ],
        "className": "bg-light-yellow",
        "style": Object {
          "zIndex": 1,
        },
      },
      "ref": [Function],
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "foo",
            "className": "block-headline center mb3",
          },
          "ref": null,
          "rendered": "foo",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
            "className": "flex w100 items-center justify-center",
          },
          "ref": null,
          "rendered": Array [],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`renders without data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AvailableLocations
    block={
        Object {
            "fields": Object {
              "contentType": "",
              "drip": false,
              "title": "",
            },
            "sys": Object {
              "id": "",
            },
          }
    }
    flavor={
        Object {
            "availableLocations": Array [],
            "contentBlocks": Array [],
            "id": "",
            "image": "",
            "label": "",
            "slug": "",
            "title": "",
          }
    }
    setRef={[Function]}
    z={1}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          className="block-headline center mb3"
/>,
        <div
          className="flex w100 items-center justify-center"
>
          
</div>,
      ],
      "className": "bg-light-yellow",
      "style": Object {
        "zIndex": 1,
      },
    },
    "ref": [Function],
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
          "className": "block-headline center mb3",
        },
        "ref": null,
        "rendered": "",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [],
          "className": "flex w100 items-center justify-center",
        },
        "ref": null,
        "rendered": Array [],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="block-headline center mb3"
/>,
          <div
            className="flex w100 items-center justify-center"
>
            
</div>,
        ],
        "className": "bg-light-yellow",
        "style": Object {
          "zIndex": 1,
        },
      },
      "ref": [Function],
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
            "className": "block-headline center mb3",
          },
          "ref": null,
          "rendered": "",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
            "className": "flex w100 items-center justify-center",
          },
          "ref": null,
          "rendered": Array [],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
